sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/CSharpSyntaxHelper.cs

```diff
@@ -267,7 +267,7 @@ public static bool IsCatchingAllExceptions(this CatchClauseSyntax catchClause)
                 return true;
             }
 
-            var exceptionTypeName = catchClause.Declaration.Type.GetText().ToString();
+            var exceptionTypeName = catchClause.Declaration.Type.GetText().ToString().Trim();

```

Comment:
because of this, "Exception " (which a space in it) was compared with "Exception" below and the code thought that some `catch` blocks weren't catching everything, so the ExitBlock was connected wrongly. This fixes this problem

Reply:
and now that I look at it, type comparison should not be done using `ToString`...